detekt-rules/src/main/kotlin/OrtImportOrder.kt

```diff
@@ -38,24 +41,100 @@ class OrtImportOrder : Rule() {
         Debt.FIVE_MINS
     )
 
+    private val commonTopLevelDomains = listOf("com", "org", "io")
+
     @ExperimentalStdlibApi
     override fun visitImportList(importList: KtImportList) {
         super.visitImportList(importList)
 
-        val importPaths = importList.imports.mapTo(mutableListOf()) { it.importPath.toString() }
-        val sortedImportPaths = importPaths.sorted().toMutableList()
-
-        // TODO: Also check for blank lines between imports from different top-level domains.
-        if (importPaths != sortedImportPaths) {
-            val path = importList.containingKtFile.absolutePath()
-            val message = "Imports in file '$path' are not sorted alphabetically"
-            val finding = CodeSmell(
-                issue,
-                // Use the message as the name to also see it in CLI output and not only in the report files.
-                Entity.from(importList).copy(name = message),
-                message
-            )
-            report(finding)
+        val children = importList.node.getChildren(null)
+        if (children.isNotEmpty()) {
+            val importPaths = children.mapNotNull {
+                if (it.elementType == KtNodeTypes.IMPORT_DIRECTIVE) {
```

Comment:
Is this `if..else if` structure not replacable with a `when (it.elementType)` ?

Reply:
Not really, since for the blank line you check `it.psi`. But maybe one can also check `when(it.psi)` and see if it is either of type `KtImportDirective` or `PsiWhiteSpace`.